### DISCUSSION TITLE
output itmdump data into a file

### DISCUSSION THREAD

**Discussion by NitinSaxenait** [japaric/discovery#389] [Q&A] [answered]
Hi all, I am working on the sensors of the Discovery Book. I am able to get the output(Readings) of sensors in the itm.txt file. Now I want to store that data in a file. How can I do that in a no_std enviourment?  
Can someone please suggest to me the way? Thanks

**Comment by sirhcel** [comment, marked as answer]
Hello @NitinSaxenait,

> I am able to get the output(Readings) of sensors in the itm.txt file. Now I want to store that data in a file. How can I do that in a no_std enviourment?

With dumping the ITM data into `itm.txt` you have written it into a file. From there you can extract the actual payload with `itmdump` (https://github.com/rust-embedded/discovery/issues/399#issuecomment-908629020). You could just pipe the output from `itmdump` into a file again
```
$ itmdump -f test.itm > payload.txt
```
and do further processing like converting it into CSV, ...


I would process ITM data on a full-blown operating system where I can use `itmdump` (which is a `std` application). What are you trying to achieve exactly? What is your use case for handling this data in a `no_std` environment?

**Reply by NitinSaxenait** [reply]
No, i tried it with a different application. I run on my code with no_std and got the readings from a file using another application. At first, I was trying to read the same itm.txt file from the single no_std environment application.

**Reply by NitinSaxenait** [reply]
@sirhcel I stopped my itmdump console and put this above command. I got another file as payload.txt with only useful data what I want. But I don't want to stop my console and covert itm.txt to payload.txt I want it to be done automatically whenever I run my sensor just like I was reading from itm.txt which was automatically updating. I there any way to achieve this?

**Reply by sirhcel** [reply]
> @sirhcel I stopped my itmdump console and put this above command. I got another file as payload.txt with only useful data what I want. But I don't want to stop my console and covert itm.txt to payload.txt I want it to be done automatically whenever I run my sensor just like I was reading from itm.txt which was automatically updating. I there any way to achieve this?

What about letting `itmdump` follow `itm.txt` and continuously update its output to `payload.txt` then? This could be achieved using the `--follow`/`-F` option:
```
$ itmdump --follow --file test.itm > payload.txt
```
This should result in `payload.txt` getting updated every time new ITM data arrives.

**Reply by NitinSaxenait** [reply]
This is great. It worked for me. I was stuck here for so long. Thanks a lot @sirhcel :)
